pool/src/transaction_pool.rs

```diff
@@ -171,9 +177,37 @@ impl TransactionPool {
 			self.adapter.tx_accepted(&entry.tx);
 		}
 
+		// Transaction passed all the checks but we have to make space for it
+		if evict {
+			self.evict_from_txpool()?;
+		}
+
 		Ok(())
 	}
 
+	// Remove the last transaction from the flattened bucket transactions.
+	// No other tx depends on it, it has low fee_to_weight and is unlikely to participate in any cut-through.
+	pub fn evict_from_txpool(&mut self) -> Result<(), PoolError> {
+		// Get bucket transactions
+		let bucket_transactions = self.txpool.bucket_transactions(Weighting::NoLimit);
+
+		// Get last transaction and remove it
+		match bucket_transactions.last() {
+			Some(evictable_transaction) => {
+				// Remove transaction
```

Comment:
We should be able use `entries.remove()` here I think.

Reply:
I'm not sure this is doable since we are comparing a `PoolEntry` with a `Transaction`.
If both were `PoolEntry` a `remove_item(...)` would be possible.

Reply:
Ah good point.